provider/source/src/units/data.rs

```diff
@@ -102,30 +106,25 @@ impl crate::IterableDataProviderCached<UnitsDisplayNameV1Marker> for SourceDataP
             length: &str,
             length_patterns: &BTreeMap<String, Patterns>,
         ) -> Result<(), DataError> {
-            let quantities = length_patterns
-                .keys()
-                .filter(|&key| {
-                    !key.starts_with(|c: char| c.is_ascii_digit())
-                        && !["per", "times", "power"]
-                            .iter()
-                            .any(|&prefix| key.starts_with(prefix))
-                            // Each unit should be prefixed with the category and a hyphen
-                            && key.contains("-")
-                })
-                .filter_map(|key| {
-                    // In order to reduce the number of units in the test data,
-                    // we only test the length-* and duration-* units.
-                    #[cfg(test)]
-                    if let Some(rest) = key.strip_prefix("length-") {
-                        Some(rest)
-                    } else if let Some(rest) = key.strip_prefix("duration-") {
-                        Some(rest)
-                    } else {
-                        None
-                    }
-                    #[cfg(not(test))]
-                    key.split_once('-').map(|(_, rest)| rest)
-                });
+            let quantities = length_patterns.keys().filter_map(|key| {
+                // In order to reduce the number of units in the test data,
+                // we only test the length-* and duration-* units.
+                #[cfg(test)]
+                if let Some(rest) = key.strip_prefix("length-") {
+                    Some(rest)
+                } else if let Some(rest) = key.strip_prefix("duration-") {
+                    Some(rest)
+                } else {
+                    None
+                }
+                #[cfg(not(test))]
+                // NOTE: any units should have the category as a prefix which is separated by `-`.
+                //       Therefore, if the `rest` is empty, it means the key is not for a unit.
```

Comment:
for `length`, `rest` will not be empty, it won't exist (`rest` will be empty for `length-`).

Reply:
why?

Reply:
I think the code works as advertised.

Reply:
the phrasing of the comment makes it sounds like `rest` is the empty string, when in fact the `rest` variable never exists if there's no `-`.

Also `rest` is called `unit` now.

Reply:
good catch, done